contracts/protocol/facets/SellerHandlerFacet.sol

```diff
@@ -94,73 +94,178 @@ contract SellerHandlerFacet is SellerBase {
-            uint256 check6 = lookups.sellerIdByClerk[_seller.clerk];
-            require(
-                (check1 == 0 || check1 == _seller.id) &&
-                    (check2 == 0 || check2 == _seller.id) &&
-                    (check3 == 0 || check3 == _seller.id) &&
-                    (check4 == 0 || check4 == _seller.id) &&
-                    (check5 == 0 || check5 == _seller.id) &&
-                    (check6 == 0 || check6 == _seller.id),
-                SELLER_ADDRESS_MUST_BE_UNIQUE
-            );
-        }
+        sellerPreUpdateChecks(_seller.id, _seller.admin, lookups);
+        sellerPreUpdateChecks(_seller.id, _seller.operator, lookups);
+        sellerPreUpdateChecks(_seller.id, _seller.clerk, lookups);
+
+        // Clean old seller pending update data if exists
+        delete lookups.sellerPendingUpdates[_seller.id];
+
+        bool needsApproval;
 
         // Admin address or AuthToken data must be present in parameters. A seller can have one or the other. Check passed in parameters
-        if (_seller.admin == address(0)) {
+        if (_authToken.tokenType != AuthTokenType.None) {
             // Check that auth token is unique to this seller
             uint256 check = lookups.sellerIdByAuthToken[_authToken.tokenType][_authToken.tokenId];
             require(check == 0 || check == _seller.id, AUTH_TOKEN_MUST_BE_UNIQUE);
+
+            delete lookups.sellerIdByAuthToken[authToken.tokenType][authToken.tokenId];
+
+            // Store auth token
+            authToken.tokenId = _authToken.tokenId;
+            authToken.tokenType = _authToken.tokenType;
+
+            // Store seller by auth token reference
+            lookups.sellerIdByAuthToken[_authToken.tokenType][_authToken.tokenId] = _seller.id;
+
+            // Remove previous admin address if it exists
+            delete lookups.sellerIdByAdmin[seller.admin];
+            delete seller.admin;
         } else {
-            // Check that the admin address is unique to one seller id across all roles -- not used or is used by this seller id.
-            uint256 check1 = lookups.sellerIdByOperator[_seller.admin];
-            uint256 check2 = lookups.sellerIdByAdmin[_seller.admin];
-            uint256 check3 = lookups.sellerIdByClerk[_seller.admin];
-            require(
-                (check1 == 0 || check1 == _seller.id) &&
-                    (check2 == 0 || check2 == _seller.id) &&
-                    (check3 == 0 || check3 == _seller.id),
-                SELLER_ADDRESS_MUST_BE_UNIQUE
-            );
+            if (_seller.admin != seller.admin) {
+                require(_seller.admin != address(0), INVALID_ADDRESS);
+                // If admin address exists, admin address owner must approve the update for prevent front-running
+                lookups.sellerPendingUpdates[_seller.id].admin = _seller.admin;
+                needsApproval = true;
+            }
+        }
+
+        if (seller.treasury != _seller.treasury) {
```

Comment:
Is there a reason for `seller` and `_seller` to be in a different order in this `If `clause?

Reply:
This one is fixed, but we still have a different order on when checking admin on line 136 and treasury on 161. Let's be consistent. This is the kind of thing the auditor will comment on as a "style" issue.

Reply:
done